dist/tools/isrvecgen/isrvecgen.py

```diff
@@ -0,0 +1,222 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Tool for automatically generating interrupt vector tables from CMSIS vendor headers
+#
+# Author:   Hauke Petersen <hauke.petersen@fu-berlin.de>
+
+import argparse
+import sys
+import os
+import glob
+import re
+import copy
+from operator import itemgetter
+
+cpulist = set()
+headermap = dict()
+
+alias = {"isr_exti[_\d+]+": "isr_exti"}
+
+def parse_cpuconf(cpuconf):
+    buf = open(cpuconf, 'r').read()
+    path = os.path.dirname(os.path.abspath(cpuconf))
+
+    pat_line = re.compile("(#if|#elif|#ifdef)(( \|\|| \|\| \\\\\n| \\\\\n +\|\|)?"
+                          " +(defined\()?CPU_MODEL_[0-9A-Z]+\)?)*"
+                          "\n#include \"(.+\.h)\"", re.MULTILINE)
+    pat_cpu = re.compile("CPU_MODEL_([0-9A-Z]+)")
+
+    for m in pat_line.finditer(buf):
+        header = '{}/{}'.format(path, m.group(5))
+
+        if header not in headermap:
+            headermap[header] = set()
+
+        for c in pat_cpu.finditer(m.group(0)):
+            cpu = c.group(1).lower()
+            cpulist.add(cpu)
+            headermap[header].add(cpu)
+
+
+def parse_header(file, cpus):
+    filename = os.path.basename(file)
+    res = {"cpus": cpus, "vec": {}};
+    vec = res["vec"]
+
+    # read through file and find all lines
+    with open(file, 'r', encoding = "ISO-8859-1") as f:
+         for line in f:
+            m = re.match(" +([_0-9A-Za-z]+)_IRQn += (-?\d+),? +"
+                         "(/\*!<|/\*\*<) (.+[a-zA-Z0-9]) +\*/", line)
+            if m:
+                num = int(m.group(2))
+                name = 'isr_{}'.format(m.group(1).lower())
+                comment = '/* [{:>2}] {} */'.format(num, m.group(4))
+
+                nm = re.match(".+reserved.+", name, re.IGNORECASE)
+                if num < 0 or nm:
+                    continue
+
+                if num in vec:
+                    sys.exit("Error: vector defined twice")
+                vec[num] = {"name": name, "comment": comment}
+
+    return res;
+
+
+def apply_alias(common, specific):
+    for num in common:
+        for a in alias:
+            if re.match(a, common[num]["name"]):
+                common[num]["name"] = alias[a]
+
+    for cl in specific:
+        for num in cl["vec"]:
+            for a in alias:
+                if re.match(a, cl["vec"][num]["name"]):
+                    cl["vec"][num]["name"] = alias[a]
+
+
+def depstr(cpus, prefix):
```

Comment:
docstring missing + try to be explicit with function names. We are not writing C code